### ISSUE TITLE
default value generator doesn't work on field collections

### ISSUE THREAD

**Issue by EmilieVilliere** [pimcore/pimcore#9078]
**Describe the bug**

Hello pimcore world, 
I have created a service that allows you to set the default date 5 years later. When I paste the path of my service in the corresponding input (in the class definition) it works fine. But when I paste it in the collection fields input it doesn't work. 
However in collection fields the input **Default value generator service** is present so we are supposed to use it. 
Does anyone have an idea? I don't understand.
Thanks for all your suggestions :)

**Comment by EmilieVilliere** [comment]
![service](https://user-images.githubusercontent.com/48009102/117680276-4b902e00-b1b1-11eb-8339-298f0dac04b8.png)
![class](https://user-images.githubusercontent.com/48009102/117680455-7aa69f80-b1b1-11eb-80a5-e81c2fc13bbf.png)

**Comment by weisswurstkanone** [comment]
@EmilieVilliere do you create the object via API oder in the backend UI?

In general, behavior should be as described here:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/index.html

Please note the comment that default values are only applied on **object creation**

**Comment by weisswurstkanone** [comment]
Hi @EmilieVilliere , closing this for now. Please feel free to open it again whenever things are not clear.

**Comment by alexanderkroneis** [comment]
Hi, I feel free to open this one again since I am facing the same issue. I implemented a `DefaultValueGenerator` and dumped it's return value and it seems that my field collection is not even calling the Default Value Generator.

Default Value Generator: `App\DefaultValueGenerator\DefaultPhoneNumberGenerator`

```
<?php

namespace App\DefaultValueGenerator;

use Pimcore\Model\DataObject\ClassDefinition\Data;
use Pimcore\Model\DataObject\ClassDefinition\DefaultValueGeneratorInterface;
use Pimcore\Model\DataObject\Concrete;

class DefaultPhoneNumberGenerator implements DefaultValueGeneratorInterface
{

    public function getValue(Concrete $object, Data $fieldDefinition, array $context): string
    {
        return '+43';
    }
}
```

**Comment by alexanderkroneis** [comment]
@dvesh3 Hi, do you know when this bug will be fixed? Our client is waiting for this feature and we can't provide it at the moment. If not, I can try to fork your repository and fix it by myself and provide a PR.

**Comment by dvesh3** [comment]
@alexgaal Hey! I have just reopened it coz i think it is a valid issue and this is not priority for us at the moment. It would be great if you can provide a PR 👍 thanks!

**Comment by mcop1** [comment]
@alexgaal any chance you had the time to look into this?

**Comment by alexanderkroneis** [comment]
No, not yet. Maybe I'll be able to take a look into it after this week since we have a really tensed schedule at the moment and this is not a priority of ours as well. 😄

**Comment by mcop1** [comment]
1.) getDataForEditMode, which handles the default values, cancels out at this line, because there is no data set. 
https://github.com/pimcore/pimcore/blob/b2f37a652b19637bf3c260de7333197d087ba2ac/models/DataObject/ClassDefinition/Data/Fieldcollections.php#L127 
2.) What are the expectations? Assuming I have an existing object and add a new FieldCollection, I would expect the default values to be set. But that won´t work because of
https://github.com/pimcore/pimcore/blob/b2f37a652b19637bf3c260de7333197d087ba2ac/models/DataObject/Traits/DefaultValueTrait.php#L56

**Comment by fashxp** [comment]
related to https://github.com/pimcore/pimcore/issues/10179

**Comment by NikitaKharkov** [comment]
Guys, any updates regarding this topic? Have the same issue.

**Comment by lukmzig** [comment]
This is the same reason as here: https://github.com/pimcore/pimcore/issues/10179#issuecomment-2346145404

@vin0401 @fashxp we should consider this behavior and tackle it when working on the studio.
